scheduler/src/cook/mesos/util.clj

```diff
@@ -363,22 +418,29 @@
                                    (-> 1 t/hours t/in-millis)))
           expanded-end (Date. (+ (.getTime end)
                                  (-> 1 t/hours t/in-millis)))
+          entid-start (d/entid-at db :db.part/user expanded-start)
+          entid-end (d/entid-at db :db.part/user expanded-end)
+          job-user-entid (d/entid db :job/user)
+          ^long start-ms (.getTime start)
+          ^long end-ms (.getTime end)
           jobs
-          (->> (d/seek-datoms db :avet :job/user user (d/entid-at db :db.part/user expanded-start))
-               (take-while #(and (< (.e %) (d/entid-at db :db.part/user expanded-end))
-                                 (= (.a %) (d/entid db :job/user))
-                                 (= (.v %) user)))
-               (map #(.e %))
+          (->> (d/seek-datoms db :avet :job/user user entid-start)
+               (take-while #(and (< (:e %) entid-end)
+                                 (= (:a %) job-user-entid)
+                                 (= (:v %) user)))
+               (map #(:e %))
                (map (partial d/entity db))
-               (filter #(<= (.getTime start) (.getTime (:job/submit-time %))))
-               (filter #(< (.getTime (:job/submit-time %)) (.getTime end)))
+               (filter #(let [^long submit-ms (.getTime (:job/submit-time %))]
+                          (and (<= start-ms submit-ms)
+                               (< submit-ms end-ms))))
                (filter #(= :job.state/completed (:job/state %))))]
       (->>
         (cond->> jobs
                  (not include-custom-executor?) (filter #(false? (:job/custom-executor %)))
                  (instance-states state) (filter #(= state (job-ent->state %)))
                  name-filter-fn (filter #(name-filter-fn (:job/name %))))
-        (take limit)))))
+        (take limit)
+        doall))))
```

Comment:
Why do we need the `doall`?

Reply:
Force the metric to not be junk. If we didn't, then this entire body is lazy and our metrics would 'improve' to about 1ms.

Reply:
https://github.com/twosigma/Cook/issues/780

Reply:
👍